test/blackbox-tests/test-cases/mdx-stanza/shared-libraries.t/dune

```diff
@@ -0,0 +1,16 @@
+(library
+ (name public_lib)
+ (foreign_archives test))
+
+(rule
+ (deps test.c)
+ (targets libtest.a dlltest.so)
+ (action
+  (progn
+   (run gcc -c -fPIC test.c -o test.o)
```

Comment:
The test should depend on the `gcc` binary and `ar` binaries.

Reply:
I fixed it in #10586